README.md

```diff
@@ -1,2 +1,112 @@
 # couchbase-cluster-manager
+
+![Swift](https://img.shields.io/badge/swift-4.0.2-orange.svg)
+![Platform](https://img.shields.io/badge/platform-OSX-lightgrey.svg)
+![Platform](https://img.shields.io/badge/platform-Linux-lightgrey.svg)
+
 Docker container for Couchbase Server cluster provisioning and configuration.
+
+## Overview
+
+Couchbase Cluster Manager is a Docker container running an application written in Swift.
+It's purpose is to automatically manage a cluster of Couchbase Server nodes running on a Docker swarm.
+
+### Features
+
+- Automatically provision a new cluster from scratch.
+- Add nodes to a cluster.
+- Remove failed nodes from a cluster.
+- Rebalance the cluster.
+
+### How it works
+
+**Glossary:**
+- `Node`: a couchbase server instance.
+- `Service`: a docker swarm service, running a couchbase server container (a node).
+
+Couchbase Cluster Manager communicate directly with Docker Swarm and retrieves at regular intervals (`CCM_INTERVAL`) a list of running services which `com.ccm.cluster` label match the manager `CCM_NAME` environment variable. It then proceeds to find an already provisioned node, if none is found a new cluster get initialized using the configuration specified on the container environment. At this stage a bucket named `CCM_BUCKET_NAME` is created, if defined in the configuration.
```

Comment:
The first sentence feels quite clumsy to read, I think it would read better as follows: `Couchbase Cluster Manager communicates directly with Docker Swarm and retrieves a list of running services at regular intervals (CCM_INTERVAL). The list is then filtered based on it's com.ccm.cluster label to match the manager's CCM_NAME environment variable.`